source/crud/tests/v2/findOneAndDelete-hint-clientError.yml

```diff
@@ -0,0 +1,43 @@
+runOn:
+  # These tests assert that the driver raises client-side errors. Server
+  # versions >= 4.1.10 raise errors for unknown findAndModify options
+  # (SERVER-40005), but the spec requires client-side errors for < 4.2. The wire
+  # version for 4.2 was bumped in 4.1.1 (SERVER-35169), so we'll allow up to
+  # 4.1.0 (inclusive).
```

Comment:
@ShaneHarvey: I think this exposes a gap in our test coverage. Server versions >= 4.1.1 will be considered 4.2 based on their wire version, but we can't actually depend on the server to report errors for unsupported `findAndModify` options until 4.1.10. This means that the `hint` option could be silently ignored if the driver is connected to a server version between 4.1.1 and 4.1.9 (inclusive).

Since version checking is typically done by wire protocol version, I don't think there's an clear solution for this short of raising client-side errors for <4.4 servers (which I don't think we want to do). We may have to concede that we can't help users running pre-release versions of the database.

It's possible this applies to other commands as well, but `findAndModify` stands out as strict option validation was introduced fairly recently. As time goes on, there will be less need to worry about this.

Reply:
I don't think we need to care about the odd-numbered releases once the even-numbered release following it has been published.  No one should be testing against any odd-numbered releases except the latest one (currently 4.3.x).  Just use 4.0 and 4.2 as the release numbers.